cbcontainers/models/compatibility_matrix_test.go

```diff
@@ -0,0 +1,58 @@
+package models_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/vmware/cbcontainers-operator/cbcontainers/models"
+)
+
+func TestCheckCompatibility(t *testing.T) {
```

Comment:
I think this should be 2 tests - one that verify compatible agents are detected and one that detects incompatible agents are detected

Reply:
the tests shared some code, that's why I made them this way. I extracted them into two top level tests, but they share a method that runs the logic